feat: Haskell language

algas opened the pull request:
## Feature

- Haskell language
`echo '{ "name": "David" }' | quicktype -l haskell`

## Test

- pass tests in my local environment
```.sh
npm run build
FEATURE=haskell npm test
```

## Others

- ~~This PR does not support GraphQL in Haskell~~
- GraphQL works with Haskell

schani commented:
@algas That's amazing!  Thank you!

What is it that keeps GraphQL from working?  There's nothing specific in the target languages about GraphQL.

algas commented:
@schani How can I test it for GraphQL files?
These code was written by reference to Elm-language code, but it did not support GraphQL.

schani commented:
Ah, you'll have to fix a bunch of lint warning.  You can locally run

`npm run tslint`

I'm actually not sure about GraphQL and Elm now.  It might be because it only supports a single top-level?  I can look into it if you'd like.

schani commented:
I haven't tried it out, but GraphQL should work with Elm.  Please check out [our blog post about GraphQL](https://blog.quicktype.io/graphql-with-quicktype/).

algas commented:
Thank you for your information about GraphQL.
I fixed lint error and added configuration for ci tests.
GraphQL works with Haskell now in some cases.

I will fix some tests failed.

schani commented:
Fantastic, thank you!

schani commented on test/fixtures/haskell/.gitignore:
```diff
@@ -0,0 +1,3 @@
+/elm-stuff
```
I'm guessing this is not needed?

algas replied:
Thank you

algas commented:
@schani What can I do to fix "buildkite/quicktype-pr/diff"?

schani commented:
I'll see if I have time to fix the diff issue on the weekend.  It's a quicktype build bug, not your bug.

schani commented:
Thank you so much for your contribution! 🙏
